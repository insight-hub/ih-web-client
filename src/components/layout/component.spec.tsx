import React from 'react';
import renderer from 'react-test-renderer';

import {
  AlignContent,
  AlignItems,
  AlignSelf,
  Background,
  Cursor,
  Display,
  FlexWrap,
  JustifyContent,
  Layout,
  LayoutProps,
  Overflow,
  Position,
  ZIndex,
} from './';

function getOptionalProps(): LayoutProps {
  return {
    position: Position.Relative,
    display: Display.Flex,
    justifyContent: JustifyContent.Around,
    flexWrap: FlexWrap.Wrap,
    alignContent: AlignContent.Around,
    alignSelf: AlignSelf.Center,
    alingItems: AlignItems.Center,
    cursor: Cursor.Pointer,
    overflow: Overflow.Scroll,
    zIndex: ZIndex.Above,
    background: Background.Base,
  };
}

describe('Layout', () => {
  it('should render correct', () => {
    const component = renderer.create(<Layout />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render correct with props', () => {
    const component = renderer.create(<Layout {...getOptionalProps()} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render witn custom css', () => {
    const instance = renderer.create(<Layout className="test-class" />).root;
    const element = instance.findByType('div');
    expect(element.props.className).toContain('test-class');
  });
});
