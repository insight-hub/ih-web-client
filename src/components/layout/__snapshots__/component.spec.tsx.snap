// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correct 1`] = `
<div
  className="sc-bcXHqe"
/>
`;

exports[`Layout should render correct with props 1`] = `
<div
  className="sc-bcXHqe lbcHjn"
  cursor="pointer"
  display="flex"
  overflow="scroll"
/>
`;
